telegram/inline/inlinequeryresult.py

```diff
@@ -41,3 +41,11 @@ def __init__(self, type, id, **kwargs):
         self.id = str(id)
 
         self._id_attrs = (self.id,)
+
+    @property
+    def _has_parse_mode(self):
+        return hasattr(self, 'parse_mode')
+
+    @property
+    def _has_input_message_content(self):
+        return hasattr(self, 'input_message_content')
```

Comment:
I dont exactly see what this has to do with the PR?

Reply:
Good question … Don't recall, why I added it …

Reply:
Mh, actually it seems that the `InlineQueryResult…` classes don't always have an `input_message_content` attribute. This is due to the fact, that the base class `InlineQueryResult` doesn't have one and in the child classes the optionals are parsed in the manner

```
if input_message_content:
            self.input_message_content = input_message_content
```
*without* preceding `self.input_message_content = None`.
I.e. if the optional `input_message_content` is not passed, the class instance won't have such an attribute.

I'm not sure that this is acutally wanted behavior, but changing that would surely be the content of another PR …

Reply:
Okay, above mentioned issue is solved by #1600. But this is still needed, since e.g. `InlineQueryResultGame` has *neither* a `parse_mode` nor ar `input_message_content` attribute.